src/aggregate/aggregate_request.c

```diff
@@ -1121,6 +1120,103 @@ static bool IsIndexCoherent(AREQ *req) {
   return true;
 }
 
+
+static int ApplyParsedVectorQuery(ParsedVectorQuery *pvq, RedisSearchCtx *sctx, QueryAST *ast, QueryError *status) {
+
+  // Resolve field spec
+  const FieldSpec *vectorField = IndexSpec_GetFieldWithLength(sctx->spec, pvq->fieldName, strlen(pvq->fieldName));
+  if (!vectorField || !FIELD_IS(vectorField, INDEXFLD_T_VECTOR)) {
+    QueryError_SetError(status, QUERY_ENOOPTION, "Vector field not found");
```

Comment:
Let's use the already existing `Unknown field` error (does not exist in `QueryErrorCode`)